params/config.go

```diff
@@ -469,6 +469,31 @@ func isForked(s, head *big.Int) bool {
 	return s.Cmp(head) <= 0
 }
 
+// isForkBlock returns whether the given block number designates a fork block
+func (c *ChainConfig) IsForkBlock(n uint64) bool {
+	number := new(big.Int).SetUint64(n)
+	for _, forkNum := range []*big.Int{
+		c.HomesteadBlock,
+		c.DAOForkBlock,
+		c.EIP150Block,
+		c.EIP155Block,
+		c.EIP158Block,
+		c.ByzantiumBlock,
+		c.ConstantinopleBlock,
+		c.PetersburgBlock,
+		c.IstanbulBlock,
+	} {
```

Comment:
Could these fork numbers be nil? Probably yes(not activated yet for some private networks). Please check the nil before compare.

Reply:
good catch, thanks